File: backend/app/models/vehicle_model.py
import uuid
from sqlalchemy import Column, String, Float, Boolean, DateTime, ForeignKey, JSON
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.db.base import Base

class Vehicle(Base):
    __tablename__ = "vehicles"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    owner_id = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False)
    type = Column(String, nullable=False) # bike, car
    brand = Column(String, nullable=False)
    model = Column(String, nullable=False)
    price_per_day = Column(Float, nullable=False)
    location = Column(String, nullable=False)
    image_url = Column(String, nullable=True)
    image_urls = Column(JSON, nullable=True)
    rating = Column(Float, default=0.0)
    lat = Column(Float, nullable=True)
    lng = Column(Float, nullable=True)
    description = Column(String, nullable=True)
    is_available = Column(Boolean, default=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    owner = relationship("User", backref="vehicles")

File: frontend/lib/shared/models/vehicle_model.dart
class Vehicle {
  final String id;
  final String ownerId;
  final String type;
  final String brand;
  final String model;
  final double pricePerDay;
  final String location;
  final bool isAvailable;
  final String? imageUrl;
  final List<String>? imageUrls;
  final double? rating;
  final String? description;
  final double? lat;
  final double? lng;

  Vehicle({
    required this.id,
    required this.ownerId,
    required this.type,
    required this.brand,
    required this.model,
    required this.pricePerDay,
    required this.location,
    required this.isAvailable,
    this.imageUrl,
    this.imageUrls,
    this.rating,
    this.description,
    this.lat,
    this.lng,
  });

  factory Vehicle.fromJson(Map<String, dynamic> json) {
    return Vehicle(
      id: json['id'] as String,
      ownerId: json['owner_id'] as String,
      type: json['type'] as String,
      brand: json['brand'] as String,
      model: json['model'] as String,
      pricePerDay: (json['price_per_day'] as num).toDouble(),
      location: json['location'] as String,
      isAvailable: json['is_available'] as bool? ?? true,
      imageUrl: json['image_url'] as String?,
      imageUrls: (json['image_urls'] as List<dynamic>?)
          ?.map((e) => e as String)
          .toList(),
      rating: (json['rating'] as num?)?.toDouble(),
      description: json['description'] as String?,
      lat: (json['lat'] as num?)?.toDouble(),
      lng: (json['lng'] as num?)?.toDouble(),
    );
  }
}
